meter/tibber-pulse.go

```diff
@@ -133,18 +122,56 @@ func NewTibberFromConfig(ctx context.Context, other map[string]interface{}) (api
 			case <-ctx.Done():
 				return
 			}
+
+			if err != nil {
+				// The pulse sometimes declines valid(!) subscription requests, and asks the client to disconnect.
+				// This invalidates the subscription, and therefore we resubscribe when exiting Run() gracefully
+				// upon server request.
+				// https://github.com/evcc-io/evcc/issues/17925#issuecomment-2621458890
+				if subErr := subscribe(t, client, cc); subErr != nil {
+					log.ERROR.Println(subErr)
+				}
+			}
 		}
 	}()
 
 	return t, nil
 }
 
+func subscribe(t *Tibber, client *graphql.SubscriptionClient, cc struct {
+	Token   string
+	HomeID  string
+	Timeout time.Duration
+}) error {
+	if t.subscriptionId != "" {
+		if err := client.Unsubscribe(t.subscriptionId); err != nil {
+			return err
```

Comment:
Log statt return? Die mag ja schon stale sein?

Reply:
Ich wäre sogar dafür, das im ersten Schritte einfach weg zu lassen. In allen Fehlern die wir kennen hat uns ja der Server schon die Subscription entzogen.